manifests/kernel_opt.pp

```diff
@@ -0,0 +1,52 @@
+# @summary Applies kernel parameter configuration via grubby
+#
+# @example Add Single Parameter
+#   grubby::kernel_opt{'keyword':}
+#
+# @example Delete a Single Parameter
+#   grubby::kernel_opt{'selinix':
+#     ensure => 'absent',
+#   }
+#
+# @example Add Parameter with Value
+#   grubby::kernel_opt{'memsize':
+#     value  => '22',
+# }
+#
+# @param opt Kernel option
+# @param ensure add or delete kernel option
+# @param value Value of kernel option
+# @param scope Which kernels to apply parameters to
+#
+define grubby::kernel_opt (
+  String[1] $opt                                                = $name,
+  Enum['present', 'absent']                             $ensure = 'present',
+  Variant[Enum['DEFAULT','ALL'],Pattern[/^TITLE=.+$/]]  $scope  = 'DEFAULT',
+  Optional[Variant[String[1],Integer]]                  $value  =  undef,
+){
+
+  $_opt = $value ? {
+    Undef   => $opt,
+    default => "${opt}=${value}",
+  }
+
+  case $ensure {
+    'present': {
+      exec { "Ensure ${_opt} kernel option is set for ${scope}":
+        command => "/sbin/grubby --update-kernel=${scope} --args=${_opt}",
+        path    => ['/bin','/usr/bin'],
+        unless  => "test `/sbin/grubby --info=${scope} | grep -c ^args=` == `/sbin/grubby --info=${scope} | grep ^args= | grep -c ${_opt}`",
+      }
+    }
+    'absent': {
+      exec { "Ensure ${_opt} kernel option is absent for ${scope}":
+        command => "/sbin/grubby --update-kernel=${scope} --remove-args=${opt}",
+        path    => ['/bin','/usr/bin'],
+        unless  => "test `/sbin/grubby --info=${scope} | grep -c ^args=` == `/sbin/grubby --info=${scope} | grep ^args= | grep -vwc ${opt}`",
```

Comment:
The changes look amazing!
There is just one small complaint from 'pdk validate' (linr that is 1 character too long in this case):
`warning: puppet-lint: manifests/kernel_opt.pp:45:140: line has more than 140 characters`
```suggestion
        unless  => "test `/sbin/grubby --info=${scope} | grep -c ^args=` == \
                   `/sbin/grubby --info=${scope} | grep ^args= | grep -vwc ${opt}`",
```

Reply:
Hi @atsonkov ,

Thank your for your response.

Indeed I changed this after last lint check -  it's a pain to match in rspec, of course I'll fix.

I was going to ask about these lines actually so maybe worth doing that before fixing linter. 

Was there a reason to do them as `test $(command) == $(command)` construction?

Would need to check a  but following is good I believe to assert if rhgb parameter is present.

```puppet
exect{'add rhgb':
   command => ....
   unless => "/sbin/grubby --info=DEFAULT | grep '^args=' | egrep -vw 'rhgb'
 }
```

In particular when I used the invalid `DEFUALT` and not `'DEFAULT'` both commads failed and so the the comparison
was true which was a bit nasty.

Steve.

Reply:
Of course guess you did that way for this case 
```
/sbin/grubby --info=ALL | grep '^args=' | grep -w rhgb
args="ro console=ttyS0,115200 console=tty0 rd.auto net.ifnames=0 crashkernel=auto rhgb quiet $tuned_params"
args="ro console=ttyS0,115200 console=tty0 rd.auto net.ifnames=0 crashkernel=auto rhgb quiet"
args="ro console=ttyS0,115200 console=tty0 rd.auto net.ifnames=0 crashkernel=auto rhgb quiet $tuned_params"
args="ro console=ttyS0,115200 console=tty0 rd.auto net.ifnames=0 crashkernel=auto rhgb quiet"
```
where rhgb might be missing from one.

Reply:
Something like this.

```bash
cat << EOF | grep '^args='  | test -z "$(grep -wv rhgb)"
args="rhgb foo"
args="rhgb bar"
EOF
```
returns 0.

while

```bash
cat << EOF | grep '^args='  | test -z "$(grep -wv rhgb)"
args="foo"
args="rhgb bar"
EOF
```

returns 1.

Reply:
#9 first